src/Microsoft.Health.Dicom.Client/DicomWebClient.Store.cs

```diff
@@ -26,35 +30,11 @@ public partial class DicomWebClient : IDicomWebClient
     {
         EnsureArg.IsNotNull(dicomFiles, nameof(dicomFiles));
 
-        var postContent = new List<Stream>();
-
-        try
-        {
-            foreach (DicomFile dicomFile in dicomFiles)
-            {
-                MemoryStream stream = GetMemoryStream();
-                await dicomFile.SaveAsync(stream).ConfigureAwait(false);
-                stream.Seek(0, SeekOrigin.Begin);
-                postContent.Add(stream);
-            }
-
-            using MultipartContent content = ConvertStreamsToMultipartContent(postContent);
-            return await StoreAsync(
-                GenerateStoreRequestUri(partitionName, studyInstanceUid),
-                content,
-                cancellationToken).ConfigureAwait(false);
-        }
-        finally
-        {
-            foreach (Stream stream in postContent)
-            {
-#if NETSTANDARD2_0
-                stream.Dispose();
-#else
-                await stream.DisposeAsync().ConfigureAwait(false);
-#endif
-            }
-        }
+        using MultipartContent content = DicomContent.CreateMultipart(dicomFiles);
```

Comment:
We're no longer allocating new buffers for each file, so we don't need to worry about their disposal